R/mod_common_forms.R

```diff
@@ -93,60 +93,236 @@ mod_common_forms_server <- function(
+        } else {
+          df
+        }
+      })
+      if (form == "Adverse events")
+        SAE_data({
+          df |>
+            dplyr::filter(grepl("Yes", `Serious Adverse Event`)) |>
+            dplyr::select(dplyr::any_of(
+              c("o_reviewed", "subject_id","form_repeat", "Name", "AESI",  "SAE Start date",
+                "SAE End date", "CTCAE severity", "Treatment related",
+                "Treatment action", "Other action", "SAE Category",
+                "SAE Awareness date", "SAE Date of death", "SAE Death reason")
+            )) |>
+            adjust_colnames("^SAE ")
+        })
     })
     
     mod_timeline_server("timeline_fig", r = r, form = form)
     
-    output[["SAE_table"]] <- DT::renderDT({
+    if (form == "Adverse events") {
+      output[["SAE_table"]] <- DT::renderDT({
+        req(form == "Adverse events")
+        datatable_custom(
+          isolate(subset(SAE_data(), input$show_all_data | subject_id == r$subject_id)), 
+          rename_vars = c("Review Status" = "o_reviewed", table_names), 
+          rownames= FALSE,
+          title = "Serious Adverse Events", 
+          escape = FALSE,
+          selection = "none",
+          callback = checkbox_callback,
+          options = list(
+            columnDefs = list(
+              list(
+                targets = "o_reviewed",
+                orderable = FALSE,
+                render = checkbox_render
+              ),
+              list(
+                targets = "subject_id",
+                visible = isolate(input$show_all_data)
+              )),
+            rowCallback = row_callback
+          ))
+      })
+      SAE_proxy <- DT::dataTableProxy("SAE_table")
+    }
+    
+    output[["common_form_table"]] <- DT::renderDT({
+      datatable_custom(
+        isolate(subset(common_form_data(), input$show_all_data | subject_id == r$subject_id)), 
+        rename_vars = c("Review Status" = "o_reviewed", table_names), 
+        rownames= FALSE,
+        title = form, 
+        escape = FALSE,
+        selection = "none",
+        callback = checkbox_callback,
+        options = list(
+          columnDefs = list(
+            list(
+              targets = "o_reviewed",
+              orderable = FALSE,
+              render = checkbox_render
+            ),
+            list(
+              targets = "subject_id",
+              visible = isolate(input$show_all_data)
+            )),
+          rowCallback = row_callback
+        ))
+    })
+    common_form_proxy <- DT::dataTableProxy('common_form_table')
+    
+    # Ensure that review data gets reset whenever the review data or subject
+    # being viewed is changed.
+    observe({
+      reload_data(reload_data() + 1)
+      session$userData$update_checkboxes[[form]] <- NULL
+      session$userData$review_records[[form]] <- data.frame(id = integer(), reviewed = character())
+    }) |> 
+      bindEvent(r$subject_id, r$review_data,
+                ignoreInit = TRUE)
+    
```

Comment:
Just wondering: do you have a specific reason why you sometimes use observe/bindevent and sometimes `observeEvent`?

Reply:
I prefer using `bindEvent()` when there are multiple triggers. In my opinion it is more readable.